client/ayon_core/hosts/houdini/plugins/publish/collect_staging_dirs_for_cleaning_up.py

```diff
@@ -0,0 +1,90 @@
+import pyblish.api
+import os
+from ayon_core.pipeline import AYONPyblishPluginMixin
+
+
+class CollectStagingDirsForCleaningUp(pyblish.api.InstancePlugin,
+                                      AYONPyblishPluginMixin):
+    """Collect Staging Directories For Cleaning Up.
+    
+    This collector collects staging directories
+    and adds them to file remove list.
+
+    CAUTION:
+        This collector deletes the parent folder of the exported files.
+        It works fine with the default filepaths in the creators.
+        Artist should be aware with that fact so they take care when 
+          changing the file path in the ROP node.
+        Developers should be aware when changing the filepath pattern
+          in creator plugins.
```

Comment:
With this huge of a caution this plug-in should NOT be enabled by default and/or it should be very specific about what files it's allowed to clean (e.g. only the expected files of an instance). If for whatever reason other files exist in the folder they should remain on disk.

This is a **huge** caution - and inheritly unsafe I'd say. We're much better of marking the folder to "delete folder only if empty" and then have something else, e.g. the extractors mark the files themselves as temporary.

Anyway, there's been quite a debate on the maya front as well with renders being deleted on publish - then the render had an issue on one frame on the farm and now suddenly you don't have all the renders there anymore. To then just fix that one frame you need to either fully re-render the shot, or manually on your server "revert the files" from backup or whatever. I'd argue that also there, the default should be "OFF" with a clear warning label **in settings** about enabling it.

Reply:
> "delete folder only if empty"
> the extractors mark the files themselves as temporary.

I believe this is safer as it doesn't force putting the staged files in separate directories.. 
How can I do that without replicating some of the logic in the integrator ?